app/src/main/java/com/douglasbruce/tasky/core/domain/validation/EmailValidator.kt

```diff
@@ -0,0 +1,21 @@
+package com.douglasbruce.tasky.core.domain.validation
+
+import javax.inject.Inject
+
+class EmailValidator @Inject constructor(
+    private val matcher: EmailMatcher
+) {
+
+    operator fun invoke(email: String): ValidationResult {
+        if(!matcher.matches(email)) {
+            return ValidationResult(
+                successful = false,
+                errorType = ErrorType.FORMAT
+            )
+        }
```

Comment:
Might also check for empty email to throw a separate error in that case

Reply:
Good point. Will add a blank check for all to provide a more specific error message for each validator